_version_helper.py

```diff
@@ -0,0 +1,97 @@
+"""
+this module is a hack only in place to allow for setuptools
+to use the attribute for the versions
+
+it works only if the backend-path of the build-system section
+from pyproject.toml is respected
+"""
+from __future__ import annotations
+
+import logging
+from typing import Callable
+
+from setuptools import build_meta as build_meta  # noqa
+
+from setuptools_scm import _types as _t
+from setuptools_scm import Configuration
+from setuptools_scm import get_version
+from setuptools_scm import git
+from setuptools_scm import hg
+from setuptools_scm.fallbacks import parse_pkginfo
+from setuptools_scm.version import (
+    get_no_local_node,
+    _parse_version_tag,
+    guess_next_simple_semver,
+    SEMVER_MINOR,
+    guess_next_version,
+    ScmVersion,
+)
+
+log = logging.getLogger("setuptools_scm")
+# todo: take fake entrypoints from pyproject.toml
+try_parse: list[Callable[[_t.PathT, Configuration], ScmVersion | None]] = [
+    parse_pkginfo,
+    git.parse,
+    hg.parse,
+    git.parse_archival,
+    hg.parse_archival,
+]
+
+
+def parse(root: str, config: Configuration) -> ScmVersion | None:
+    for maybe_parse in try_parse:
+        try:
+            parsed = maybe_parse(root, config)
+        except OSError as e:
+            log.warning("parse with %s failed with: %s", maybe_parse, e)
```

Comment:
should be `maybe_parse.__name__`?

Reply:
for `git.parse` maybe parse will give more info like
```
<function parse at 0x7fa03a40e5f0>
```

but __name__ would be just 
```
parse
```